app/assets/javascripts/modules/map/leaflet-plugins/granule_grid_layer.js.coffee

```diff
@@ -37,12 +38,22 @@ ns.GranuleGridLayer = do (
 
   class GranuleGridLayer extends L.GridLayer
     initialize: (@collection, color, @multiOptions) ->
+      console.warn 'collection', @collection
       if @collection.granuleDatasource()
         @granules = @collection.granuleDatasource().data()
       else
         @_datasourceSubscription = @collection.granuleDatasource.subscribe(@_subscribe)
 
-      @color = color ? '#25c85b';
+      # If on the search page, the color should always be green.
+      if page.current.page() == 'search'
+        color = '#25c85b'
+
+      # If were on the project page, set it to blue while we finish loading the collection data.
+      if page.current.page() == 'project'
```

Comment:
This doesn't seem overly bad but we should ensure we are making this check consistently -- I've seen similar things before checking the URL, is this a better way of making this check?

Reply:
This was something I've introduced as a base class for pages. That seems like a bit cleaner, more maintainable way to do it, rather than relying on an href. I like that it ties into Knockout-land a little better. Interested to hear any thoughts if you have any.

Reply:
👍